django/http/request.py

```diff
@@ -51,6 +51,21 @@ class HttpRequest:
     _encoding = None
     _upload_handlers = []
 
+    non_picklable_attrs = frozenset(
+        [
+            "resolver_match",
+            "environ",
+            "META",
+            "_stream",
+        ]
+    )
+    META_non_picklable_attr = frozenset(
+        [
+            "wsgi.errors",
+            "wsgi.input",
+        ]
+    )
```

Comment:
`wsgi.errors` and `wsgi.input` are specific for `WSGIRequest` so we should move this to the `WSGIRequest.__getstate__()`. It seems that `META` is picklable for `HttpRequest` and `ASGIRequest`.

Reply:
Ok will do it 👍🏻  and yes it seems `META` is picklable for `HttpRequest` and `ASGIRequest`.

Reply:
@felixxm so the WSGIRequest.__getstate__() and WSGIRequest.__deepcopy__() would look something like this :
```py
non_picklable_attrs = frozenset(
        [
            "wsgi.errors",
            "wsgi.input",
        ]
    )
    def __getstate__(self):
        obj_dict = self.__dict__.copy()
        for attr in self.non_picklable_attrs:
            if attr in obj_dict["META"]:
                del obj_dict["META"][attr]
        super().__getstate__()

    def __deepcopy__(self, memo):
        obj = copy.copy(self)
        for attr in self.non_picklable_attrs:
            if hasattr(self, attr):
                setattr(obj, attr, copy.deepcopy(getattr(self, attr), memo))
        memo[id(self)] = obj
        super().__deepcopy__(memo)
```